src/main/java/nl/novi/TechItEasy/services/TelevisionService.java

```diff
@@ -88,6 +96,22 @@ public Television assignCIModuleToTelevision(int televisionId, long ciModuleId){
         return television;
     }
 
+    public void assignRemoteControllerToTelevision(int televisionId, long remoteControllerId){
+        Television television = televisionRepository.findById(televisionId).orElseThrow(() -> new RecordNotFoundException(televisionId + " not found"));
+        RemoteController remoteController = remoteControllerRepository.findById(remoteControllerId).orElseThrow(() -> new RecordNotFoundException(remoteControllerId + " not found"));
+
+        television.setRemoteController(remoteController);
```

Comment:
Ik zie dat je hier niet eerst checkt of de remotecontroller wel aanwezig is. Dit kan problemen veroorzaken mocht er een keer geen remotecontroller aanwezig zijn/ verkeerd id nummer wordt ingevuld. Ik zie dat je dat wel in je toTelevisionDto doet maar die gebruik je niet in de methode voor zover ik zie.

Reply:
Die check zit er wel in. Dat is de ".orElseThrow()" methode. Dit is een methode van Optional die checkt of er iets in de Optional zit, zo niet dan gooit het de opgegeven exception op (dit staat in een lambda functie, zodat de orElseThrow methode het kan aanroepen/uitvoeren).

Dit is in feite het zelfde als de check die we geleerd hebben met een if statement met .isEmpty() of .isPresent(). Deze manier is eigenlijk "syntactic sugar" voor zo'n if-statment. We behandelen in de les liever niet syntactic sugar, zodat je expliciet leert wat er gebeurt in de logica